src/Phinx/Db/Action/ChangeColumn.php

```diff
@@ -72,11 +72,12 @@ public function __construct(Table $table, $columnName, Column $column)
      * @param mixed $options Additional options for the column
      * @return ChangeColumn
      */
-    public static function build(Table $table, $columnName, $type = null, $options = [])
+    public static function build(Table $table, $columnName, $type = null, $options = [], $adapter = null)
```

Comment:
This is not ok, options should be passed to the object and not fetched here.

Reply:
when a column change is performed, if the null option not specified it will replace the existing value with the default value from Phinx\Db\Action\Column.php class.

Currently, am fetching existing options of the column from Database in Phinx\Db\Action\Column.php class (isn't it right??). Can I fetch the existing options of the column from Database in Phinx/Db/Action/ChangeColumn.php

Reply:
No, the part of fetching data inside this class is the part I'm objecting to. You should fetch it from the outside and pass the data as an argument

Reply:
Got it @lorenzo, For confirmation, can I fetch the existing options of the column from Database in Phinx/Db/Action/ChangeColumn.php?

Reply:
No, you should not fetch the from that class, they should be passed as arguments.